src/engraving/draw/pen.h

```diff
@@ -0,0 +1,147 @@
+    Pen(const QBrush &brush, double width, PenStyle s = SolidLine,
+        PenCapStyle c = SquareCap, PenJoinStyle j = BevelJoin) :
+    _brush(brush), _width(width), _style(s), _capStyle(c), _joinStyle(j) {}
+    
+    PenStyle style() const { return _style; }
+    void setStyle(PenStyle style) {
+        if (_style == style)
+            return;
+        _style = style;
+        _dashPattern.clear();
+      }
+    
+    QVector<double> dashPattern() const
+    {
+        if (_style == SolidLine || _style == NoPen) {
+            return QVector<qreal>();
+        } else if (_dashPattern.isEmpty()) {
+            const qreal space = 2;
+            const qreal dot = 1;
+            const qreal dash = 4;
+            switch (_style) {
+            case DashLine:
+                _dashPattern.reserve(2);
+                _dashPattern << dash << space;
+                break;
+            case DotLine:
+                _dashPattern.reserve(2);
+                _dashPattern << dot << space;
+                break;
+            case DashDotLine:
+                _dashPattern.reserve(4);
+                _dashPattern << dash << space << dot << space;
+                break;
+            case DashDotDotLine:
+                _dashPattern.reserve(6);
+                _dashPattern << dash << space << dot << space << dot << space;
+                break;
+            default:
+                break;
+            }
+        }
+        return _dashPattern;
+    }
+    
+    void setDashPattern(const QVector<double> &pattern)
+    {
+        if (pattern.isEmpty())
+            return;
+        _dashPattern = pattern;
+        _style = CustomDashLine;
+        if ((_dashPattern.size() % 2) == 1) {
+            LOGW() << "Pattern not of even length";
+            _dashPattern << 1;
+        }
+    }
+    
+    double widthF() const { return _width; }
+    void setWidthF(double width) { _width = width; }
+
+
+    int width() const { return qRound(_width); }
+    void setWidth(int width) { _width = width; }
+
+    QColor color() const { return _brush.color(); }
+    void setColor(const QColor &color) { _brush = QBrush(color); }
+    QBrush brush() const { return _brush; }
+    void setBrush(const QBrush &brush) { _brush = brush; }
+    bool isSolid() const { return _brush.style() == Qt::SolidPattern; }
+    PenCapStyle capStyle() const { return _capStyle; }
+    void setCapStyle(PenCapStyle pcs) { _capStyle = pcs; }
+    PenJoinStyle joinStyle() const { return _joinStyle; }
+    void setJoinStyle(PenJoinStyle pcs) { _joinStyle = pcs; }
+    
+#ifndef NO_QT_SUPPORT
+    QPen toQPen() const { return QPen(_brush, _width, static_cast<Qt::PenStyle>(_style), static_cast<Qt::PenCapStyle>(_capStyle), static_cast<Qt::PenJoinStyle>(_joinStyle)); }
+#endif
+
+private:
+    void setDefaults() {
+        _brush = Qt::black;
+        _width = _defaultWidth;
+        _style = _penDefaultStyle;
+        _capStyle = _penDefaultCap;
+        _joinStyle = _penDefaultJoin;
+    }
```

Comment:
I believe that this method is unnecessary and members like `defaultX` are not needed
It is sufficient (and necessary) to simply initialize the fields with their default values.
```
double _width = 1.0;
PenStyle _style = SolidLine;
...
```